src/server/game/AI/ScriptedAI/ScriptedEscortAI.cpp

```diff
@@ -472,34 +481,38 @@ void npc_escortAI::Start(bool isActiveAttacker /* = true*/, bool run /* = false
     m_bCanInstantRespawn = instantRespawn;
     m_bCanReturnToStart = canLoopPath;
 
+    if (!ScriptWP && resetWaypoints) // sd2 never adds wp in script, but tc does
+        FillPointMovementListForCreature();
+
     if (m_bCanReturnToStart && m_bCanInstantRespawn)
         TC_LOG_DEBUG("scripts", "EscortAI is set to return home after waypoint end and instant respawn at waypoint end. Creature will never despawn.");
 
     if (me->GetMotionMaster()->GetCurrentMovementGeneratorType() == WAYPOINT_MOTION_TYPE)
     {
-        me->GetMotionMaster()->MovementExpired();
+        me->StopMoving();
+        me->GetMotionMaster()->Clear(false);
         me->GetMotionMaster()->MoveIdle();
         TC_LOG_DEBUG("scripts", "EscortAI start with WAYPOINT_MOTION_TYPE, changed to MoveIdle.");
     }
 
     //disable npcflags
-    me->SetUInt32Value(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_NONE);
+    me->SetFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_NONE);
```

Comment:
This line does nothing.
Old code removed all npc flags
Your new code appends "NONE" flag (no effect)